java/fury-core/src/main/java/org/apache/fury/resolver/ClassResolver.java

```diff
@@ -514,22 +513,21 @@ public List<Class<?>> getRegisteredClasses() {
    * a class is registered but not an inner class with inner serializer, it will still be taken as
    * non-final to write class def, so that it can be deserialized by the peer still..
    */
-  public boolean isFinal(Class<?> clz) {
-    if (Modifier.isFinal(clz.getModifiers())) {
-      if (fury.getConfig().shareMetaContext()) {
-        boolean isInnerClass = isInnerClass(clz);
-        if (!isInnerClass) {
-          return false;
-        } else {
-          // can't create final map/collection type using TypeUtils.mapOf(TypeToken<K>,
-          // TypeToken<V>)
-          return !Map.class.isAssignableFrom(clz) && !Collection.class.isAssignableFrom(clz);
-        }
+  public boolean isMonomorphic(Class<?> clz) {
+    if (fury.getConfig().shareMetaContext()) {
+      if (!ReflectionUtils.isMonomorphic(clz)) {
+        return false;
+      }
+      boolean isInnerClass = isInnerClass(clz);
+      if (!isInnerClass) {
+        return false;
       } else {
-        return true;
+        // can't create final map/collection type using TypeUtils.mapOf(TypeToken<K>,
+        // TypeToken<V>)
+        return !Map.class.isAssignableFrom(clz) && !Collection.class.isAssignableFrom(clz);
       }
```

Comment:
Can it be simplified to `return ReflectionUtils.isMonomorphic(clz) && isInnerClass(clz) && (!Map.class.isAssignableFrom(clz) && !Collection.class.isAssignableFrom(clz))` ?

Reply:
updated